terraform/aws/analytical-platform-management-production/terraform-state/dynamodb-table.tf

```diff
@@ -12,3 +12,94 @@ import {
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${var.account_ids["analytical-platform-management-production"]}:role/GlobalGitHubActionAdmin"]
+    }
+  }
+  // Analytical Platform Team
+  statement {
+    sid    = "AnalyticalPlatformTeam"
+    effect = "Allow"
+    actions = [
+      "dynamodb:DescribeTable",
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:DeleteItem"
+    ]
+    resources = [module.state_locking.dynamodb_table_arn]
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${var.account_ids["analytical-platform-management-production"]}:role/aws-reserved/sso.amazonaws.com/${data.aws_region.current.name}/${one(data.aws_iam_roles.analytical_platform_team_access_role.names)}"]
+    }
+  }
+  // Data Engineering: Data Engineering Production
+  statement {
+    sid    = "DataEngineeringProduction"
+    effect = "Allow"
+    actions = [
+      "dynamodb:DescribeTable",
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:DeleteItem"
+    ]
+    resources = [module.state_locking.dynamodb_table_arn]
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${var.account_ids["analytical-platform-data-engineering-production"]}:role/aws-reserved/sso.amazonaws.com/${data.aws_region.current.name}/${one(data.aws_iam_roles.data_engineering_team_access_role_data_engineering_production_data_eng.names)}"]
+    }
+  }
+  // Data Engineering: Data Engineering Sandbox A
+  statement {
+    sid    = "DataEngineeringSandboxA"
+    effect = "Allow"
+    actions = [
+      "dynamodb:DescribeTable",
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:DeleteItem"
+    ]
+    resources = [module.state_locking.dynamodb_table_arn]
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:aws:iam::${var.account_ids["analytical-platform-data-engineering-sandbox-a"]}:role/aws-reserved/sso.amazonaws.com/${data.aws_region.current.name}/${one(data.aws_iam_roles.data_engineering_team_access_role_data_engineering_sandbox_a_admin.names)}",
+        "arn:aws:iam::${var.account_ids["analytical-platform-data-engineering-sandbox-a"]}:role/aws-reserved/sso.amazonaws.com/${data.aws_region.current.name}/${one(data.aws_iam_roles.data_engineering_team_access_role_data_engineering_sandbox_a_data_eng.names)}"
+      ]
+    }
+  }
+  // Data Engineering: Data Production
+  statement {
+    sid    = "DataProduction"
+    effect = "Allow"
+    actions = [
+      "dynamodb:DescribeTable",
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:DeleteItem"
+    ]
+    resources = [module.state_locking.dynamodb_table_arn]
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${var.account_ids["analytical-platform-data-production"]}:role/aws-reserved/sso.amazonaws.com/${data.aws_region.current.name}/${one(data.aws_iam_roles.data_engineering_team_access_role_data_production_data_eng.names)}"]
```

Comment:
Yeah so data engineers will be able to plan from data eng account or data prod account. I want that too! Gimme that!!!